### ISSUE TITLE
Working-ish

### ISSUE THREAD

**Issue by xylo04** [n8jja/Pat-Vara#5]
I have a branch where connections are half-working. I need a place to stash this stack trace before I lose it.

This occurred at https://github.com/xylo04/Pat-Vara/commit/5221f0d539b917ad7925bbb0a10b47ada187b61d plus https://github.com/xylo04/pat/commit/2c502e445b8ae99c07546c6a12ac5dfd3de64875

It's close! :smile: @n8jja and @martinhpedersen FYI! Oh, and if you're using Pat's RMS list, it's currently showing center frequencies instead of dial frequencies, so dial down 1.5 kHz.

```
2021/09/07 16:19:57 QSY vara: 7102.3
2021/09/07 16:20:00 Connecting to WA5TED (vara)...
2021/09/07 16:20:28 Connected to WA5TED (vara)
RMS Trimode 1.3.40.2 Welcome  to the Lubbock ARES Winlink Gateway - hosted by the Lubbock 9-1-1 Dist,
K0SWE has 1440 minutes remaining with WA5TED (DM93AN)
{SFI = 100 On 2021-09-07 22:00 UTC}
[WL2K-5.0-B2FWIHJM$]
;PQ: 21379644
CMS via WA5TED >
>FF
2021/09/07 16:22:19 got a vara command I wasn't expecting: BUFFER 62
2021/09/07 16:22:32 got a vara command I wasn't expecting: BUFFER 34
2021/09/07 16:22:39 got a vara command I wasn't expecting: BUFFER 0
FQ
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5dd6d9]

goroutine 41 [running]:
github.com/n8jja/Pat-Vara/vara.(*Modem).cmdListen(0xc0004f0000)
        /home/chris/src/Pat-Vara/vara/vara.go:99 +0x59
created by github.com/n8jja/Pat-Vara/vara.(*Modem).start
        /home/chris/src/Pat-Vara/vara/vara.go:69 +0xd2
```

**Comment by n8jja** [comment]
So when you say working-ish, does that mean you can send/receive with it? Or just contact and connect to the gateway?

Jeremy

**Comment by xylo04** [comment]
At #6 I was able to make a connection and do one exchange session, but usually the disconnect would screw up the state and I'd have to restart Pat in order to try again. As of #10 I can repeatedly make outgoing connections. The remaining debate is around correctness compared to Winlink Express' interactions with VARA, and about listening for incoming connections instead of just making outgoing connections.

**Comment by xylo04** [comment]
If you'd like to try it out, see https://github.com/xylo04/pat/tree/vara!

**Comment by xylo04** [comment]
Heads up, this repo is now officially merged as a dependency of Pat! Pat v0.13.0 will feature official VARA support! https://github.com/la5nta/pat/pull/361
